project_admin/tests/tests_views.py

```diff
@@ -0,0 +1,55 @@
+from django.test import TestCase, Client
+from django.core.management import call_command
+from django.conf import settings
+
+
+class AdminLoginTestCase(TestCase):
+    """
+    Tests Admin login
+    """
+
+    def setUp(self):
+        """
+        Set up the app for following tests.
+        """
+        settings.DEBUG = True
+        settings.ADMIN_PASSWORD = 'test1234'
+        call_command('init_proj_config')
+
+    def test_admin_login_redirect(self):
+        """
+        Test redirection to the login page
+        """
+        c = Client()
+        response = c.get("/project-admin/")
+        self.assertRedirects(response, '/project-admin/login')
+
+    def test_admin_login_success(self):
+        """
+        Test successful log in attempt.
+        """
+        c = Client()
+        response = c.post("/project-admin/login/",
+                          {'password': 'test1234'},
+                          follow=True)
+        self.assertRedirects(response, '/project-admin/')
+        self.assertContains(response, "Current Project Configuration")
+
+    def test_admin_login_fail(self):
+        """
+        Test unsuccessful login attempt.
+        """
+        c = Client()
+        response = c.post("/project-admin/login/", {'password': 'meep'})
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Password incorrect")
+
+    def test_project_config_page_returns_correct_html(self):
```

Comment:
Sorry, but this function is still redundant. `test_admin_login_success` tests basically the same thing. The only difference is that you also test for the template that is used. Please just add the template assertion to the already existing function so that we don't have multiple tests doing the same thing.

Reply:
Sorry for being so late in fixing this. I'll take care that there's no duplication of efforts from next time.